plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeMinioConnectorTest.java

```diff
@@ -955,6 +956,242 @@ public void testSettingChangeDataFeedEnabledProperty()
+        assertUpdate(
+                format("INSERT INTO " + tableName + " VALUES (1, ROW('%s', 3)), (11, ROW('%1$s', 13)), (21, ROW('%1$s', 23)), (31, ROW('%1$s', 33))", largeVarchar),
+                4);
+
+        String selectQuery = "SELECT a.c FROM " + tableName;
+        Set<Integer> expected = ImmutableSet.of(3, 13, 23, 33);
+        Session sessionWithoutPushdown = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), "projection_pushdown_enabled", "false")
+                .build();
+
+        assertQueryStats(
+                getSession(),
+                selectQuery,
+                statsWithPushdown -> {
+                    DataSize processedDataSizeWithPushdown = statsWithPushdown.getProcessedInputDataSize();
+                    assertQueryStats(
+                            sessionWithoutPushdown,
+                            selectQuery,
+                            statsWithoutPushdown -> assertThat(statsWithoutPushdown.getProcessedInputDataSize()).isGreaterThan(processedDataSizeWithPushdown),
+                            results -> assertEquals(results.getOnlyColumnAsSet(), expected));
+                },
+                results -> assertEquals(results.getOnlyColumnAsSet(), expected));
+
+        assertUpdate("DROP TABLE IF EXISTS " + tableName);
+    }
+
+    @Test
+    public void testProjectionPushdownOnPartitionedTables()
+    {
+        String tableName1 = "test_table_with_partition_at_beginning_" + randomNameSuffix();
+        String tableName2 = "tes_table_with_partition_at_end_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName1 + " (id BIGINT, root ROW(f1 BIGINT, f2 BIGINT)) WITH (partitioned_by = ARRAY['id'])");
+        assertUpdate("INSERT INTO " + tableName1 + " VALUES (1, ROW(1, 2)), (1, ROW(2, 3)), (1, ROW(3, 4))", 3);
+        assertQuery("SELECT id, root.f2 FROM " + tableName1, "VALUES (1, 2), (1, 3), (1, 4)");
+
+        assertUpdate("CREATE TABLE " + tableName2 + " (root ROW(f1 BIGINT, f2 BIGINT), id BIGINT) WITH (partitioned_by = ARRAY['id'])");
+        assertUpdate("INSERT INTO " + tableName2 + " VALUES (ROW(1, 2), 1), (ROW(2, 3), 1), (ROW(3, 4), 1)", 3);
+        assertQuery("SELECT root.f2, id FROM " + tableName2, "VALUES (2, 1), (3, 1), (4, 1)");
+
+        assertUpdate("DROP TABLE " + tableName1);
+        assertUpdate("DROP TABLE " + tableName2);
+    }
+
+    @Test
+    public void testProjectionPushdownMultipleRows()
+    {
+        String tableName = "test_projection_pushdown_multiple_rows_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName +
+                " (id BIGINT, nested1 ROW(child1 BIGINT, child2 VARCHAR, child3 INT), nested2 ROW(child1 DOUBLE, child2 BOOLEAN, child3 DATE))");
+        assertUpdate("INSERT INTO " + tableName + " VALUES" +
+                        " (100, ROW(10, 'a', 100), ROW(10.10, true, DATE '2023-04-19'))," +
```

Comment:
I would also add a NULL ROW and ROWs with NULLs in some sub-fields for more variety